### ISSUE TITLE
Insecure rpath in binary and embedded ruby

### ISSUE THREAD

**Issue by ronisaacson** [pmq20/ruby-packer#66]
After building a binary with rubyc, the binary contains an insecure rpath. The ruby interpreter in the embedded squashfs also contains the same insecure rpath.

```
% readelf -d a.out | grep RPATH
 0x000000000000000f (RPATH)              Library rpath: [/tmp/rubyc/ruby-2.4.1-0.4.0/build/lib]

% readelf -d /tmp/rubyc/rubyc_work_dir/__enclose_io_memfs__/bin/ruby | grep RPATH
 0x000000000000000f (RPATH)              Library rpath: [/tmp/rubyc/ruby-2.4.1-0.4.0/build/lib]
```

These both use shared libraries. An attacker could easily create /tmp/rubyc/ruby-2.4.1-0.4.0/build/lib and put a malicious libc.so.6 in there, for example, compromising any other user running a rubyc-built binary on that system.

**Comment by kke** [comment]
This is work-aroundable by using the `-d` parameter to set the temporary path during builds. (You don't have to rebuild rubyc)

Maybe if you set it to `/root` or why not `/__enclose_io_memfs__` it's more secure.

**Comment by ronisaacson** [comment]
Well, sort of, but then you have to run your build as root. This is not generally recommended, and you're still building in an unnecessary and insecure path. A better option would be to include --disable-rpath in the configure options.

**Comment by kke** [comment]
Yes, but it's still better than /tmp until an option like that is available. You could build inside a chroot jail or a docker container without root.
